source/SH3/graphics/texture.cpp

```diff
@@ -13,48 +13,330 @@
 #include <SH3/arc/vfile.hpp>
 
 #include <cassert>
+#include <fstream>
 
+#define HEADER_START 64
+
+static std::size_t offset = 0;
 
 using namespace sh3_graphics;
 
+#pragma pack(push, 1)
+struct rgba
+{
+    std::uint8_t r, g, b, a;
+};
+
+struct rgba16
+{
+    uint8_t r : 5;
+    uint8_t g : 5;
+    uint8_t b : 5;
+    uint8_t a : 1;
+};
+
+struct rgb
+{
+    std::uint8_t r, g, b;
+};
+
+struct bgra
+{
+    std::uint8_t b, g, r, a;
+};
+
+struct bgr
+{
+    std::uint8_t b, g, r;
+};
+
+struct tga_header
+{
+    std::uint8_t id_size;   // Size of the ID field that follows this header (usually 0)
+    std::uint8_t colormap;  // Is this image paletted (THIS MUST ALWAYS BE 0 FOR US!)
+    std::uint8_t type;      // Type of image (ALWAYS 2 FOR RGB24!)
+    std::uint8_t unused[5]; // Colormap crap
+
+    std::uint16_t x_origin;  // Co-ordinate for first X value
+    std::uint16_t y_origin;  // Co-ordinate for first Y value
+    std::uint16_t width;
+    std::uint16_t height;
+    std::uint8_t bpp;       // Bits per pixel
+    std::uint8_t flags;
+};
+
+#pragma pack(pop)
+
+/**
+ *  Dump a texture to a Microsoft BMP/DIB
+ *
+ *  @param width - The Width of this texture.
+ *  @param height - The Height of this texture.
+ *  @param data - Pixel data
+ */
+void Dump2Bitmap(std::uint32_t width, std::uint32_t height, std::vector<std::uint8_t>& data)
+{
+    tga_header header;
+    std::ofstream file("output.tga", std::ios::binary); // Open the stream for binary output
+
+    memset(&header, 0, sizeof(header));
+
+    header.id_size      = 0x00;
+    header.colormap     = 0x00;
+    header.type         = 2;
```

Comment:
I think a class-scoped static constant is good.

Since many of these fields have sane default values, you could also use member initialization in the class definition like so:
```
struct tga_header
{
    static constexpr std::uint8_t TYPE_RGB24 = 2;

    std::uint8_t id_size = 0;
    std::uint8_t colormap = 0;
    std::uint8_t type = TYPE_RGB24;
    ...
};
```
So we don't need the `memset` or many of the assignments here.

We could also list all possible types in an enum, but if we're only going to use RGB24 anyway it's not too useful. I wouldn't necessarily be against it though.